src/protocol/http/src/test/java/org/apache/jmeter/gui/action/TestCorrelationExtractor.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed  under the  License is distributed on an "AS IS" BASIS,
+ * WITHOUT  WARRANTIES OR CONDITIONS  OF ANY KIND, either  express  or
+ * implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jmeter.gui.action;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.jmeter.protocol.http.gui.action.CorrelationExtractor;
+import org.apache.jmeter.samplers.SampleResult;
+import org.junit.Test;
+
+public class TestCorrelationExtractor {
+
+
+    @SuppressWarnings("deprecation")
+    @Test
+    public void testcreateExtractor() {
+
+        Map<String, String> bodyParameterMap = new HashMap<>();
+        bodyParameterMap.put("_csrf", "7d1de481-34af-4342-a9b4-b8288c451f7c");
+
+        List<String> arguments = new ArrayList<>();
+        arguments.add("_csrf");
+
+        SampleResult sampleResult = new SampleResult();
+        sampleResult.setResponseData("<html><head><title>Login Page</title></head><body onload='document.f.username.focus();'>"
```

Comment:
Does this test data need to be so large?
Could some of the tags or tag attributes or text be reduced/removed to make the purpose of the test clearer?

Reply:
Ham1-san.

> I think this functionality is really valuable for JMeter, I've just made a few comments only 
> based upon quickly reading the code, with the aim of trying to make it easier to read :)

Thank you for comments. And I'm happy to interest my pull requests.
Our team check the comments and deal it.